test/plugin/test_filter_process_snmptrap.rb

```diff
@@ -42,36 +39,53 @@ def filter(records, conf = CONFIG)
   def test_snmptrap_filter
     records = [
         {
-                "SNMPv2-MIB::sysUpTime.0":"43 days, 21:28:56.10","SNMPv2-MIB::snmpTrapOID.0":"SNMPv2-SMI::enterprises.59.3.800.100.20.3","SNMPv2-SMI::enterprises.59.3.800.10.10.1.1":"UV300-00000550","SNMPv2-SMI::enterprises.59.3.800.10.30.1.1":"r001i01b","SNMPv2-SMI::enterprises.59.3.800.30.10.1.1":"SYSPOWERSTATE","SNMPv2-SMI::enterprises.59.3.800.30.10.1.2":"0x0","SNMPv2-SMI::enterprises.59.3.800.30.10.1.4":"18","host":"192.168.254.61"
```

Comment:
Was this format with the ":" not valid?

Reply:
tl;dr. Lets just say that with ":" didn't work, when I changed it, it start working. =)

The long story seems to be related to how the Hash is created on the feed() above. It takes a string, and magically convert it to a Hash.. For some reason, if the string is created with ":", the record hash in the filter plugin won't find the key unless I prepend the key with ":" (e.g. record[:"SNMPv2-SMI::enterprises.59.3.800.30.10.1.1"). But if you do that, the filter plugin will not find the key when it's loaded by fluentd (instead of the unit test engine).
I hear you asking WHAT? CAN WE CREATE INCOMPATIBLE HASHES DEPENDING ON THE SINTAX? WHY? and I will answer you, because Ruby.
[https://stackoverflow.com/questions/8675206/is-there-any-difference-between-the-key-value-and-key-value-hash-no]()

Reply:
Ok, but do the real SNMP traps we see from the RMC contain ":"?

Reply:
Not sure what you mean, SNMP is a binary protocol, when we see a trap, we just see 1 and 0s. They mostly represent a sequence of numbers that you need to translate. You can read/format them as you wish.

If your question is how the snmp input plugin send it to the filter plugin, it's sent as a Ruby Hash. On how the input plugin works, it uses a Ruby library that translates the traps and creates a Hash from them.

Reply:
Oh, ok.